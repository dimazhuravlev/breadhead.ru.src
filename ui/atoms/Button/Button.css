@import "../../../pages/config.css";

.Button {
  background-color: var(--beigeSecondary);
  border-radius: 50px;
  padding: 4px 20px;
  box-sizing: border-box;
  outline: none;
  border: none;
  display: flex;
  align-items: center;
  cursor: pointer;
  &:hover {
    background-color: var(--redPrimary);
    transition: all 0.2s ease-in;
  }
}

.ButtonText {
  color: var(--darkPrimary);
  font: var(--caption20-24);
  margin-bottom: 3px;
}

.ButtonIcon {
  margin-right: 8px;
}
