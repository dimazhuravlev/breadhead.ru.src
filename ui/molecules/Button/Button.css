@import "../../../pages/config.css";

.Button {
  background-color: var(--beigeSecondary);
  border-radius: 50px;
  padding: 4px 20px;
  box-sizing: border-box;
  outline: none;
  border: none;
  display: flex;
  align-items: center;
  cursor: pointer;
  &:hover {
    background-color: var(--redPrimary);
    transition: all 0.15s ease-in;
  }
  &:active {
    background-color: var(--redSecondary);
  }
}

.ButtonText {
  color: var(--darkPrimary);
  font: var(--caption20-24);
  margin-bottom: 3px;
}

.Button svg {
  margin-right: 8px;
}
