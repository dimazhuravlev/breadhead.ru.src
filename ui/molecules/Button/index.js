import React from 'react'
import PropTypes from 'prop-types'
import styles from './Button.css'
import cx from 'classnames'

const Button = React.forwardRef((props, ref) => (
  <button
    style={props}
    ref={ref}
    name={props.name}
    onClick={props.onClick}
    className={cx(styles.Button, props.className)}
  >
    {props.icon}
    <p className={styles.ButtonText}>{props.children}</p>
  </button>
))

Button.propTypes = {
  children: PropTypes.string,
  icon: PropTypes.object
}

export default Button
