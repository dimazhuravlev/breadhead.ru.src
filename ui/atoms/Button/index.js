import React from 'react'

import { ZipIcon } from '@site/ui/molecules/Icons'

import styles from './Button.css'

const Button = () => (
  <button className={styles.Button}>
    <ZipIcon className={styles.ButtonIcon} />
    <div className={styles.ButtonText}>cвязь</div>
  </button>
)

export default Button
