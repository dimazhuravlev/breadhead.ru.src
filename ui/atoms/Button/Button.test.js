import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Button from '.'

describe('button ui', () => {
  test('Button renders correctly', () => {
    const output = shallow(<Button />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
