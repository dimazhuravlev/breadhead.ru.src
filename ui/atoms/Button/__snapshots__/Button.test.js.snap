// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button ui Button renders correctly 1`] = `
<button>
  <ZipIcon />
  <div>
    cвязь
  </div>
</button>
`;
